src/Configurator/AddLinesConfigurator.php

```diff
@@ -23,7 +23,61 @@ class AddLinesConfigurator extends AbstractConfigurator
         self::POSITION_AFTER_TARGET,
     ];
 
+    /**
+     * Holds file contents for files that have been loaded.
+     * This allows us to "change" the contents of a file multiple
+     * times before we actually write it out.
+     *
+     * @var string[]
+     */
+    private $fileContents = [];
+
     public function configure(Recipe $recipe, $config, Lock $lock, array $options = []): void
+    {
+        $this->fileContents = [];
+        $this->executeConfigure($recipe, $config);
+
+        foreach ($this->fileContents as $file => $contents) {
+            $this->write(sprintf('[add-lines] Patching file "%s"', $this->relativize($file)));
+            file_put_contents($file, $contents);
+        }
+    }
+
+    public function unconfigure(Recipe $recipe, $config, Lock $lock): void
+    {
+        $this->fileContents = [];
+        $this->executeUnconfigure($recipe, $config);
+
+        foreach ($this->fileContents as $file => $change) {
+            $this->write(sprintf('[add-lines] Reverting file "%s"', $this->relativize($file)));
+            file_put_contents($file, $change);
+        }
+    }
+
+    public function update(RecipeUpdate $recipeUpdate, array $originalConfig, array $newConfig): void
+    {
+        // manually check for "requires", as unconfigure ignores it
+        $originalConfig = array_filter($originalConfig, function ($item) {
+            return !isset($item['requires']) || $this->isPackageInstalled($item['requires']);
+        });
+
+        // reset the file content cache
+        $this->fileContents = [];
+        $this->executeUnconfigure($recipeUpdate->getOriginalRecipe(), $originalConfig);
+        $this->executeConfigure($recipeUpdate->getNewRecipe(), $newConfig);
```

Comment:
The old logic "unconfigured" the original recipe, then configured the new recipe. So we're doing the same, except that we "record" the changes and report them to `$recipeUpdate` below instead of actually *making* the changes now.